src/strategy/dungeons/wotlk/forgeofsouls/FosMultipliers.cpp

```diff
@@ -0,0 +1,55 @@
+#include "FosMultipliers.h"
+#include "FosActions.h"
+#include "GenericSpellActions.h"
+#include "ChooseTargetActions.h"
+#include "MovementActions.h"
+#include "FosTriggers.h"
+#include "FosActions.h"
+
+
+float BronjahmMultiplier::GetValue(Action* action) {
+    Unit* boss = nullptr;
+    boss = AI_VALUE2(Unit *, "find target", "bronjahm");
+    if (!boss)
+        return 1.0f;
+    if (botAI->IsTank(bot))
+        if (dynamic_cast<TankAssistAction*>(action))
+            return 0.0f;
+
+    if (boss && boss->HasUnitState(UNIT_STATE_CASTING) && boss->FindCurrentSpellBySpellId(SPELL_CORRUPT_SOUL))
+    {
+        if (dynamic_cast<MovementAction*>(action) && !dynamic_cast<MoveFromBronjahmAction*>(action))
+        {
+            return 0.0f;
+        }
+    }
+    return 1.0f;
+}
+
+float AttackFragmentMultiplier::GetValue(Action* action)
+{
+    if (botAI->IsHeal(bot) && dynamic_cast<AttackCorruptedSoulFragmentAction*>(action))
```

Comment:
Logic should be moved to triggers, `AttackCorruptedSoulFragmentAction` should never be called for healers and you shouldn't need to suppress it in a multiplier